# **Complete EDA On Google Playstore Apps**

## About Dataset

This is google playstore apps dataset available at [Kaggle](https://www.kaggle.com/lava18/google-play-store-apps)


`Context`

While many public datasets (on Kaggle and the like) provide Apple App Store data, there are not many counterpart datasets available for Google Play Store apps anywhere on the web. On digging deeper, I found out that iTunes App Store page deploys a nicely indexed appendix-like structure to allow for simple and easy web scraping. On the other hand, Google Play Store uses sophisticated modern-day techniques (like dynamic page load) using JQuery making scraping more challenging.

`Content`
Each app (row) has values for catergory, rating, size, and more.

`Acknowledgements`

This information is scraped from the Google Play Store. This app information would not be available without it.

`Inspiration`

The Play Store apps data has enormous potential to drive app-making businesses to success. Actionable insights can be drawn for developers to work on and capture the Android market!

## 1. Importing Libraries

In [1]:
import pandas as pd
import numpy as np
import seaborn as sns

import matplotlib.pyplot as plt
%matplotlib inline

**ignore warnings**

In [2]:
import warnings
warnings.filterwarnings("ignore")

## 2.0 Data Loading, Filtering And Cleaning

. Load the csv file with the pandas

. creating the dataframe and understanding the data present in the dataset using pandas

. Dealing with the missing data, outliers and the incorrect records

In [3]:
df = pd.read_csv(r"D:\OneDrive\Desktop\data-analysis-course-preply\lectures\lecture-03-Google-Playstore-EDA\googleplaystore.csv")

In [4]:
df.head()

,a,Category,Rating,Reviews,Size,Installs,Type,Price,Content Rating,Genres,Last Updated,Current Ver,Android Ver,Unnamed: 13
0,Photo Editor & Candy Camera & Grid & ScrapBook,ART_AND_DESIGN,4.1,159,19M,"10,000+",Free,0,Everyone,Art & Design,"January 7, 2018",1.0.0,4.0.3 and up,NaN
1,Coloring book moana,ART_AND_DESIGN,3.9,967,14M,"500,000+",Free,0,Everyone,Art & Design;Pretend Play,"January 15, 2018",2.0.0,4.0.3 and up,NaN
2,"U Launcher Lite – FREE Live Cool Themes, Hide ...",ART_AND_DESIGN,4.7,87510,8.7M,"5,000,000+",Free,0,Everyone,Art & Design,"August 1, 2018",1.2.4,4.0.3 and up,NaN
3,Sketch - Draw & Paint,ART_AND_DESIGN,4.5,215644,25M,"50,000,000+",Free,0,Teen,Art & Design,"June 8, 2018",Varies with device,4.2 and up,NaN
4,Pixel Draw - Number Art Coloring Book,ART_AND_DESIGN,4.3,967,2.8M,"100,000+",Free,0,Everyone,Art & Design;Creativity,"June 20, 2018",1.1,4.4 and up,NaN


checking columns in the dataset

In [5]:
df.columns

Index(['a', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type',
       'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver',
       'Android Ver', 'Unnamed: 13'],
      dtype='str')

### checking number of rows and columns

In [6]:
df.shape

(10841, 14)

### checking what kind of data types we have in the dataset

In [7]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 10841 entries, 0 to 10840
Data columns (total 14 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   a               10841 non-null  str    
 1   Category        10840 non-null  str    
 2   Rating          9367 non-null   float64
 3   Reviews         10841 non-null  int64  
 4   Size            10841 non-null  str    
 5   Installs        10841 non-null  str    
 6   Type            10840 non-null  str    
 7   Price           10841 non-null  str    
 8   Content Rating  10841 non-null  str    
 9   Genres          10840 non-null  str    
 10  Last Updated    10841 non-null  str    
 11  Current Ver     10833 non-null  str    
 12  Android Ver     10839 non-null  str    
 13  Unnamed: 13     0 non-null      float64
dtypes: float64(2), int64(1), str(11)
memory usage: 1.2 MB


In [9]:
df.describe()

,Rating,Reviews,Unnamed: 13
count,9367.000000,1.084100e+04,0.0
mean,4.191513,4.441119e+05,NaN
std,0.515735,2.927629e+06,NaN
min,1.000000,0.000000e+00,NaN
25%,4.000000,3.800000e+01,NaN
50%,4.300000,2.094000e+03,NaN
75%,4.500000,5.476800e+04,NaN
max,5.000000,7.815831e+07,NaN


## Observation 1

-> We have 10841 and 13 columns in the dataset

-> We have columns of different data types

-> The columns in the dataset are: 

`'a', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type',
       'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver',
       'Android Ver', 'Unnamed: 13'`

-> There are some missing values in the dataset, we'll handle them latter

-> There are some columns present which should be numeric but are not numeric

`Size` , `Price` , `Installs`

### **Converting Size Column To Numeric**

In [10]:
# check if size column has null values or not
df['Size'].isnull().sum()

np.int64(0)

we don't have null values, we are good to go

In [12]:
pd.set_option('display.max_columns', None) # this is to display all the columns in the dataframe
pd.set_option('display.max_rows', None) # this is to display all the rows in the dataframe

In [23]:
# checking unique values inside Size columns
df['Size'].unique()

array([1.99229440e+07, 1.46800640e+07, 9.12261120e+06, 2.62144000e+07,
       2.93601280e+06, 5.87202560e+06, 3.04087040e+07, 3.46030080e+07,
       3.25058560e+06, 2.93601280e+07, 1.25829120e+07, 2.09715200e+07,
       2.20200960e+07, 3.87973120e+07, 2.83115520e+06, 5.76716800e+06,
       1.78257920e+07, 4.08944640e+07, 3.25058560e+07, 4.40401920e+06,
       7.34003200e+06, 2.41172480e+07, 6.29145600e+06, 6.39631360e+06,
       4.82344960e+06, 9.64689920e+06, 5.45259520e+06, 1.15343360e+07,
       2.51658240e+07,            nan, 9.85661440e+06, 1.57286400e+07,
       1.04857600e+07, 1.25829120e+06, 2.72629760e+07, 8.38860800e+06,
       8.28375040e+06, 5.87202560e+07, 5.97688320e+07, 3.67001600e+07,
       5.66231040e+07, 2.05824000e+05, 3.77487360e+06, 5.97688320e+06,
       9.01775360e+06, 2.51658240e+06, 2.83115520e+07, 2.62144000e+06,
       1.67772160e+07, 3.56515840e+06, 9.33232640e+06, 4.08944640e+06,
       3.04087040e+06, 3.98458880e+07, 3.35544320e+07, 5.66231040e+06,
      

In [ ]:
# find the values inside Size column that has 'M' in it
df['Size'].loc[df['Size'].str.contains('M')].value_counts().sum()
  

np.int64(8830)

In [17]:
# find the values inside Size column that has 'K' in it
df['Size'].loc[df['Size'].str.contains('k')].value_counts().sum()

np.int64(316)

In [19]:
# find the values inside Size column that has Varies with device in it
df['Size'].loc[df['Size'].str.contains('Varies with device')].value_counts().sum()

np.int64(1695)

In [20]:
# Total values inside size column
df['Size'].value_counts().sum()

np.int64(10841)

## Observation 2

`We have 8830 values in M units`

`We have 316 values in k units`

`We have 1695 value in Varies with device`

In [ ]:
# step 1: WE will  covert 'M'=  10^6 and K = 10^3 into bytes and varies with device as well 
# And then convert them into numeric

def convert_size(size):
    
    if isinstance(size, str): #isinstance(object, type)  -> True, if the object is of that type
        
        if 'k' in size:   # chcek if size is a string like size = '19M' And then isinstance(size, str) returns True
            return float(size.replace('k', '')) * 1024
        # size.replace -> 
        elif 'M' in size:
            return float(size.replace('M', '')) * 1024 * 1024
        elif 'Varies with device' in size:
            return np.nan  #NUMPY
    return size

df['Size'] = df['Size'].apply(convert_size)
    

In [24]:
# rename the column name 'Size' to 'Size_in_bytes'
df.rename(columns={'Size': 'Size_in_bytes'}, inplace=True)

In [ ]:
# making a new column called 'Size in Mb' which will have the size in MB

df['Size_in_Mb'] = df['Size_in_bytes'].apply(lambda x: x/(1024*1024))

## Assignment: Get idea about Lamda Function
## You will deal with Install Column and Convert String values to numeric
